src/display.cpp

```diff
@@ -89,29 +89,27 @@ namespace {
 unsigned int display::zoom_ = DefaultZoom;
 unsigned int display::last_zoom_ = SmallZoom;
 
-// Factored out of set_zoom() so it can be re-used in constructor (loading from preferences)
-// new_zoom is the zoom factor (corresponding to display::zoom_)
-// new_zoom_index is the index of zoom_levels corresponding to zoom (corresponding to display::zoom_index_)
-// Returns true when validated and and zoom_index set and false if zoom cannot be validated (in which case parameters remain unchanged)
-static bool validate_zoom_level_and_set_index(unsigned int& new_zoom, int& new_zoom_index)
+// Returns index of zoom_levels which is closest match to input tile_size
+// Assumption: zoom_levels is a sorted vector of ascending tile sizes
+// (game_config is set-up this way as of October 2019 but there is no documentation stating it has to be so)
```

Comment:
Instead of adding a comment here that the assumption is undocumented, please add a comment to game_config documenting that the value must be set in ascending order :)